lib/home/instance_info_dialog.dart

```diff
@@ -1,55 +1,176 @@
+import 'dart:io';
+
 import 'package:collection/collection.dart';
+import 'package:data_size/data_size.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:intl/intl.dart';
 import 'package:lxd/lxd.dart';
 import 'package:lxd_service/lxd_service.dart';
+import 'package:provider/provider.dart';
 import 'package:ubuntu_service/ubuntu_service.dart';
 
+import '../workshops_l10n.dart';
+import 'instance_info_model.dart';
+
 Future<void> showInstanceInfoDialog(
-    {required BuildContext context, required String instanceName}) async {
+    {required BuildContext context, required String instanceName}) {
   final l10n = AppLocalizations.of(context);
-
-  final state = await getService<LxdService>().getInstanceState(instanceName);
-  final interfaces = Map<String, LxdInstanceNetworkState>.from(state.network!)
-    ..removeWhere((interface, _) => interface == 'lo');
-
-  return showDialog(
-    context: context,
-    builder: (context) => AlertDialog(
-      title: Text(l10n.instanceInformation),
-      content: Table(
-        defaultColumnWidth: const IntrinsicColumnWidth(),
-        children: [
-          TableRow(
-            children: [
-              Text(l10n.networkInterface),
-              const Text('IPv4'),
-              const Text('IPv6')
-            ]
-                .map((e) => Padding(padding: const EdgeInsets.all(8), child: e))
-                .toList(),
+  final model = InstanceInfoModel(
+      instanceName: instanceName, service: getService<LxdService>());
+  return model.init().then((_) => showDialog(
+        context: context,
+        builder: (context) => AlertDialog(
+          title: Text(l10n.instanceInformationTitle),
+          content: ChangeNotifierProvider(
+            create: (_) => model,
```

Comment:
i see why model.init() is done before showing the dialog but i'm wondering if considered nullable initial state?

https://pub.dev/packages/provider#reusing-an-existing-object-instance

Reply:
the remaining issue is that the model is not disposed because we're creating it by hand. either we need to manually dispose it ourselves after the dialog has been closed or let the provider take care of creation and disposal. i'm leaning towards the latter and using a nullable initial state because asynchronously late-initialized non-nullable members can be dangerous if anything unexpected happens during the initialization. we'd need to make the dialog handle the initial null state gracefully, though. we can make the dialog size predictable if that helps to avoid resizing when it gets populated.